kcidb/test_oo.py

```diff
@@ -0,0 +1,224 @@
+                        "origin": "kernelci"
+                    },
+                    {
+                        "build_id": "redhat:redhat.org:619c65f970"
+                                    "9de72e90f2efd0",
+                        "id": "redhat:redhat.org:b9d8be63bc2abca63165"
+                              "de5fd74f0f6d2f0b0d1c",
+                        "origin": "redhat",
+                        "waived": True,
+                        "status": "DONE"
+                    },
+                ]
+            }
+        )
+        self.client = kcidb.oo.Client(source)
+
+    def test_client_variable(self):
+        """Check instance for client variable"""
+        self.assertIsInstance(self.client, kcidb.oo.Client)
+
+    def test_one_object(self):
+        """Check that one object is returned for every type"""
+
+        self.assertIsTestInstance(self.query_str(
+            ">test[kernelci:kernelci.org:619c656de1fb4af479f2efaa]#"),
+            self.one
+        )
+
+        self.assertIsBuildInstance(self.query_str(
+            ">build[kernelci:kernelci.org:619c6a30636e8fe042f2efa0]#"),
+            self.one
+        )
+
+        self.assertIsCheckoutInstance(self.query_str(
+            ">checkout[_:redhat:5acb9c2a7bc836e9619c65f9709de72e90f2efd0]#"),
+            self.one
+        )
+
+        self.assertIsRevisionInstance(self.query_str(
+            '>revision["5acb9c2a7bc836e9e5172bbcd2311499c5b4e5f1", ""]#'),
+            self.one
+        )
+
+    def test_two_object(self):
+        """Check that two objects are returned for every type"""
+
+        self.assertIsTestInstance(self.query_str(
+            ">test[kernelci:kernelci.org:619c656de1fb4af479f2efaa;"
+            "redhat:redhat.org:b9d8be63bc2abca63165de5fd74f0f6d2f0b0d1c]#"),
+            self.two
+        )
+
+        self.assertIsBuildInstance(self.query_str(
+            ">build[redhat:redhat.org:619c65f9709de72e90f2efd0;"
+            "kernelci:kernelci.org:619c6c8d764403e3d4f2efff]#"),
+            self.two
+        )
+
+        self.assertIsCheckoutInstance(self.query_str(
+            ">checkout[_:kernelci:5acb9c2a7bc836e9e5172bbcd2311499c5b4e5f1;"
+            "_:redhat:5acb9c2a7bc836e9619c65f9709de72e90f2efd0]#"),
+            self.two
+        )
+
+        self.assertIsRevisionInstance(self.query_str(
+            '>revision["5acb9c2a7bc836e9e5172bbcd2311499c5b4e5f1", "";'
+            '"bd355732283c23a365f7c55206c0385100d1c389", ""]#'),
+            self.two
+        )
+
+    def test_misc(self):
+        """Check that the correct instance is returned from the OO Client."""
+
+        # check that empty query string returns empty object.
+        self.assertEqual(self.query_str(""), {})
+
+        # Check that one object of every type is returned together.
+        self.assertContainsAllSchemaType(self.query_str(
+            '>revision["bd355732283c23a365f7c55206c0385100d1c389", ""]#'
+            '>checkout#>build#>test#'), self.one
+        )
```

Comment:
And, as another example, this could be rewritten as:
```python
self.assertResponseContains(
    '>revision["bd355732283c23a365f7c55206c0385100d1c389", ""]#'
    '>checkout#>build#>test#',
    {"revision", "checkout", "build", "test"}, 1
)
```